src/Utilities/TupleSlice.hpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions for slicing tuples
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+namespace tuple_impl_detail {
+
+template <size_t Start, typename Tuple, size_t... Is>
+constexpr auto slice_impl(Tuple&& tuple,
+                          std::index_sequence<Is...> /*meta*/) noexcept {
+  return std::tuple<std::tuple_element_t<Is + Start, std::decay_t<Tuple>>...>{
+      std::get<Is + Start>(std::forward<Tuple>(tuple))...};
+}
+
```

Comment:
why the blank line here and right after the namespace? That's 25% of this code chunk going to blank lines

Reply:
The blank lines make the code more readable I think 🤷‍♂️ Sometimes when reading through code I miss where namespaces begin and end, and the blank lines help to visually indicate that

Reply:
I thought that's what the `namespace` keyword and ending comment are for :D I think in this case it's a bit excessive since 25% of the lines are blank...

Reply:
fine :)